tests/ui/significant_drop_in_scrutinee.stderr

```diff
@@ -11,163 +14,205 @@ LL ~     let value = mutex.lock().unwrap().foo();
 LL ~     match value {
    |
 
-error: temporary with significant drop in match scrutinee
+error: temporary with drop impl with side effects in match scrutinee lives to end of block
   --> $DIR/significant_drop_in_scrutinee.rs:132:11
    |
 LL |     match s.lock_m().get_the_value() {
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL |     }
+   |      - temporary lives until here. If mutex is locked within block, deadlocks can occur.
    |
 help: try moving the temporary above the match
    |
 LL ~     let value = s.lock_m().get_the_value();
 LL ~     match value {
    |
 
-error: temporary with significant drop in match scrutinee
+error: temporary with drop impl with side effects in match scrutinee lives to end of block
   --> $DIR/significant_drop_in_scrutinee.rs:153:11
    |
 LL |     match s.lock_m_m().get_the_value() {
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL |     }
+   |      - temporary lives until here. If mutex is locked within block, deadlocks can occur.
    |
 help: try moving the temporary above the match
    |
 LL ~     let value = s.lock_m_m().get_the_value();
 LL ~     match value {
    |
 
-error: temporary with significant drop in match scrutinee
+error: temporary with drop impl with side effects in match scrutinee lives to end of block
   --> $DIR/significant_drop_in_scrutinee.rs:201:11
    |
 LL |     match counter.temp_increment().len() {
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL |     };
+   |      - temporary lives until here. If mutex is locked within block, deadlocks can occur.
```

Comment:
The message could be confusing when there's no mutexes involved. It would be good to be able to say

> if `counter` is a mutex that is locked within the block, deadlocks can occur

Or something to those effects, if getting the name is tricky

Reply:
Agreed. See my above suggestion https://github.com/rust-lang/rust-clippy/pull/8981#discussion_r897665627 Don't mention mutexes at all, but say what the result might be. Which is "deadlocks or other unexpected behavior".